### PR TITLE
DDF-1145: Added null check to setPolicies() method to prevent NullPointerException and added extra logging statements.

### PR DESCRIPTION
**Pull request by lessarderic**
@pklinef, @stustison, please review.
@figliold, you're my hero :-)

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/95)

### CODE REVIEW

**Comment by figliold** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

<sup>**[platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java, line 122 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/95#-Jw8nISU6bpXoh_jKzvv)** ([raw file](https://github.com/codice/ddf/blob/38fd17d236d29ccfefe8bf34502b45596a84f796/platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java#L122)):</sup>
Should this be logged at debug instead?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/95)

**Comment by lessarderic** [comment]
<sup>**[platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java, line 122 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/95#-Jw8nISU6bpXoh_jKzvv)** ([raw file](https://github.com/codice/ddf/blob/38fd17d236d29ccfefe8bf34502b45596a84f796/platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java#L122)):</sup>
Since this should rarely happen in a deployed system, I thought info was more appropriate, but if you feel otherwise I can change that.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/95)

**Comment by stustison** [comment]
:+1:

File: platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java [hunk 1]
```diff
@@ -104,7 +104,27 @@ public void setContextPolicy(String path, ContextPolicy contextPolicy) {
...
+     * details on how and when this method may be called.
+     *
+     * @param properties map of properties to use to initialize the policy store.
+     *                   Since there is no configuration file bound to these properties by default,
+     *                   this map may be {@code null}.
+     */
     public void setPolicies(Map<String, Object> properties) {
+        if (properties == null) {
+            LOGGER.info("setPolicies() called with null properties map. "
+                    + "Policy store should have already been initialized so ignoring.");
```
Comment (pklinef) [inline platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java]: Based on the content of this log message, I think this should be at a debug level.

**Comment by pklinef** [comment]
:+1:

**Comment by figliold** [comment]
Built and all tests passed :thumbsup: 

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/95)

**Pull request merged**